File: backend/app/api/v1/comments.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.db.session import get_db
from app.services import comment_service
from app.schemas.comment import CommentCreate, CommentOut

router = APIRouter(prefix="/tickets/{ticket_id}/comments", tags=["Comments"])

@router.get("", response_model=list[CommentOut])
def list_comments(ticket_id: str, db: Session = Depends(get_db)):
    return comment_service.list(db, ticket_id)

@router.post("", response_model=CommentOut, status_code=201)
def create_comment(ticket_id: str, data: CommentCreate, db: Session = Depends(get_db)):
    return comment_service.create(db, ticket_id, data)

File: frontend/src/api/comments.js
import api from "./axios";

export const getComments = async (ticketId) => {
  const res = await api.get(`/tickets/${ticketId}/comments`);
  return res.data;
};

export const createComment = async (ticketId, data) => {
  const res = await api.post(`/tickets/${ticketId}/comments`, data);
  return res.data;
};
